cmd/build.js

```diff
@@ -0,0 +1,186 @@
+const Localdrive = require('localdrive')
+const Hyperdrive = require('hyperdrive')
+const Corestore = require('corestore')
+const fs = require('bare-fs')
+const os = require('bare-os')
+const path = require('bare-path')
+const { print, outputter, InputError } = require('./iface')
+const subsystem = require('../lib/subsystem')
+const { LOCALDEV, SWAP, CHECKOUT, PLATFORM_CORESTORE } = require('../lib/constants')
+const parse = require('../lib/parse')
+
+class RequirementError extends Error {
+  constructor (message) {
+    super(message)
+    this.name = 'RequirementError'
+  }
+}
+
+/* global Bare */
+module.exports = (ipc) => async function (args) {
+  try {
+    const { _: [passedKey, buildDirArg], verbose } = parse.args(args, { boolean: ['verbose'], alias: { verbose: 'v' } })
+
+    const key = parse.runkey(passedKey)?.key?.z32
+    if (!key) {
+      throw new InputError(passedKey ? `Key "${passedKey}" is not valid` : 'No key provided')
+    }
+
+    print(`Building application ${key} for ${os.platform()}-${os.arch()}...`)
+
+    // TODO: Cleanup this workaround once pear-dev no longer relies on process
+    global.process = require('bare-process')
+    global.process.getuid = () => 1000
+    global.process.getgid = () => 1000
+
+    const drive = await createPlatformDrive()
```

Comment:
is there a reason this is being done in the client? 
 
if not, then it needs to be implemented as  a method called `build` on the `Engine` class in lib/sidecar.js (pear-rpc branch). 

When the method is called `build` needs to load the system from the platform drive, which is on the engine instance (`this.drive`) 

You also need to add `build` to https://github.com/holepunchto/pear-ipc/blob/main/methods.js - it'll then be auto-added to the ipc instance so you can call it here as ipc.build(opts) 

See if you can also unload the subsystem when the client closes (in the build method on the Engine class in lib/sidecar.js)

Reply:
@davidmarkclements Hmm.. I'm thinking that this could be problematic on Windows where the build process needs to execute in a specific environment (x64 Native Tools Command Prompt for VS 2022) otherwise it would not have the required commands such as cmake in its %PATH%.

If I understand it correctly, moving the build function to Engine would make it execute in sidecar which could potentially be running in a different environment (if it was previously started by running Keet etc.). It might lead to issues such as users trying to run pear build on the native tools command prompt where they have cmake but it keeps saying cmake not found.

Reply:
Update: this has now been removed as we've moved the call to the sidecar